### PR TITLE
Improve object enumeration performance

### PR DESCRIPTION
**Pull request by bbrto21**
Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

**Comment by bbrto21** [comment]
octane 1264->1301

File: src/runtime/Object.cpp [hunk 1]
```diff
@@ -896,12 +896,20 @@ ObjectHasPropertyResult Object::hasIndexedProperty(ExecutionState& state, const
...
+    }
+};
+
 template <typename ResultType, typename ResultBinder>
 static ResultType objectOwnPropertyKeys(ExecutionState& state, Object* self)
 {
     // https://www.ecma-international.org/ecma-262/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys
     struct Properties {
-        VectorWithInlineStorage<32, std::pair<Value::ValueIndex, ObjectStructurePropertyDescriptor>, GCUtil::gc_malloc_allocator<std::pair<Value::ValueIndex, ObjectStructurePropertyDescriptor>>> indexes;
+        std::priority_queue<IndexItem, std::vector<IndexItem, GCUtil::gc_malloc_allocator<IndexItem>>, CompareIndexItem> indexes;
```
Comment (ksh8281) [inline src/runtime/Object.cpp:912]: https://stackoverflow.com/questions/5895792/why-is-using-a-stdmultiset-as-a-priority-queue-faster-than-using-a-stdpriori
Could you test another stl class?

Comment (bbrto21) [reply src/runtime/Object.cpp:912]: OK, I'll test and compare multiset and priority queue.

Comment (bbrto21) [reply src/runtime/Object.cpp]: As per your suggestion, I tried using several containers in stl.
Here are averages after running octane 10 times.

  | As-is | priority_queue | set | multiset
-- | -- | -- | -- | --
average | 1379.05 | 1399.46 | 1392.28 | 1442.88

I don't understand this result clearly, but multset was the most effective anyway. so I updated this PR using `multset`.
If any of you are interested, please double-check this.

**Review by clover2123 — APPROVED** [review]
LGTM 👍

File: src/runtime/Object.cpp (hunk 1 above)
Comment (ksh8281) [reply src/runtime/Object.cpp]: Good!

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**